File: leetcode/go/2586.count-the-number-of-vowel-strings-in-range.go
func vowelStrings(words []string, left int, right int) int {
    count := 0
    for i:=left ; i < right+1; i++ {
        if isVowelString(words[i]) {
            count++
        }
    }
    return count
}

func isVowelString(word string) bool {
    f, l := word[0], word[len(word)-1]
    if ((f == 'a' || f == 'e' || f == 'i' || f == 'o' || f == 'u') && (l == 'a' || l == 'e' || l == 'i' || l == 'o' || l == 'u')) {
        return true
    }
    return false
}

File: leetcode/rust-leetcode/src/2586.count-the-number-of-vowel-strings-in-range.rs
impl Solution {
    pub fn is_vowel_string(word: String) -> i32 {
        let wb = word.into_bytes();
        let (f, l) = (wb[0] as char, wb[wb.len()-1] as char);
        if (f == 'a' || f == 'e' || f == 'i' || f == 'o' || f == 'u') && (l == 'a' || l == 'e' || l == 'i' || l == 'o' || l == 'u') {
            1
        } else {
            0
        }
    }
    pub fn vowel_strings(words: Vec<String>, left: i32, right: i32) -> i32 {
        (left..=right).map(|i| Solution::is_vowel_string(words[i as usize].clone())).sum()
    }
}

